docs/architecture/adr-006-poap-v2.md

```diff
@@ -0,0 +1,75 @@
+# ADR 006: POAP v2 Contract
+
+## Changelog
+
+- Jul 11, 2023: Initial draft;
+
+## Status
+PROPOSED Not Implemented
+
+## Abstract
+This ADR defines the new architecture of a POAP v2 Smart Contract that will replace the current `cw721-poap`, `poap-manager`, and `poap` smart contracts.
+
+## Context
+The name POAP stands for Proof of Attendance Protocol, which is a simple NFT that attests to the fact that a user has participated in a particular event.
+
+Currently, we have implemented the ability to create and manage POAPs with three smart contracts: `cw721-poap`, `poap-manager`, and `poap`. Although this division works and makes sense under the Single Responsibility Principle, there are several downsides to it. First, users can easily upload and instantiate multiple instances of each of those contracts, leading to an unnecessary amount of contracts on the chain. Second, it makes it harder for newcomers to understand the interactions between the three contracts. Lastly, having multiple smart contracts for something as simple as a POAP seems excessive, as ideally, a POAP should have minimal operations.
+
+## Decision
+The decision is to merge all POAP-related operations into a single smart contract called `poap`, replacing all existing POAP-related smart contracts.
+
+## Specifications
+Below are the specifications for the various messages that the `poap` contract should support.
+
```

Comment:
Do we want to make it compatible with cw721?
If this is the case, we should provide all its interfaces.
https://github.com/CosmWasm/cw-nfts/blob/main/packages/cw721/README.md

Reply:
Yes, the idea was actually implementing the `cw721` interface by extending the `cw721-base` contract. I have now added this inside the specifications section.